src/connect/components/form/add-remote-drive-form.tsx

```diff
@@ -1,43 +1,45 @@
 import {
-    AddDriveInput,
     AvailableOfflineToggle,
     Disclosure,
     Divider,
     DriveName,
     FormInput,
     Label,
     LocationInfo,
+    PUBLIC,
+    SharingType,
+    SWITCHBOARD,
 } from '@/connect';
 import { Button, Icon } from '@/powerhouse';
 import { requestPublicDrive } from 'document-drive/utils/graphql';
 import { useEffect, useState } from 'react';
 import { useForm } from 'react-hook-form';
 import { useDebounceValue } from 'usehooks-ts';
 
-interface PublicDriveDetails extends AddDriveInput {
+type RemoteDriveDetails = {
     id: string;
-    driveName: string;
-    sharingType: 'PUBLIC' | 'SHARED';
-    location: 'SWITCHBOARD';
+    name: string;
+    sharingType: SharingType;
+    location: typeof SWITCHBOARD;
     availableOffline: boolean;
-}
+};
 
 type Inputs = {
     availableOffline: boolean;
 };
 
-export type AddPublicDriveInput = PublicDriveDetails & { url: string };
+export type AddRemoteDriveInput = RemoteDriveDetails & { url: string };
 
 type AddPublicDriveFormProps = {
-    sharingType?: 'PUBLIC' | 'SHARED';
-    onSubmit: (data: AddPublicDriveInput) => void;
+    sharingType: SharingType;
+    onSubmit: (data: AddRemoteDriveInput) => void;
     onCancel: () => void;
 };
 
-export function AddPublicDriveForm(props: AddPublicDriveFormProps) {
-    const { sharingType = 'PUBLIC' } = props;
-    const [publicDriveDetails, setPublicDriveDetails] =
-        useState<PublicDriveDetails>();
+export function AddRemoteDriveForm(props: AddPublicDriveFormProps) {
```

Comment:
I changed the name of this because the form actually handles public drives _and_ cloud drives